Source/WebCore/style/ChildChangeInvalidation.cpp

```diff
@@ -70,7 +70,7 @@ void ChildChangeInvalidation::invalidateForChangedElement(Element& changedElemen
         checkingContext.matchesAllHasScopes = true;
 
         for (auto* selector : invalidationRuleSet.invalidationSelectors) {
-            if (isFirst) {
+            if (isFirst && invalidationRuleSet.isNegation == IsNegation::No) {
```

Comment:
I think we could maintain this optimization in the negation case by reversing the test. Can be considered separately.